Installation issue: hip with cmake 3.21.3

### Steps to reproduce the issue



```console
$ spack install hip@4.3.2 ^cmake@3.21.3

     56      -- Found HIP: /tmp/user/spack-stage/spack-stage-hip-4.3.1-tlzwetxef5vw4roytm6wdhoeep3m5vcw/spack-build-tlzwetx (found version "4.3.0")
     57      -- host interface - found
     58      -- HIP runtime lib type - shared
     59      -- CMAKE_TESTING_TOOL:
     60      -- Configuring done
  >> 61      CMake Error in CMakeLists.txt:
     62        HIP_ARCHITECTURES is empty for target
     63        "directed_tests.kernel.hipLaunchParm".
     64      
     65      
  >> 66      CMake Error in CMakeLists.txt:
     67        HIP_ARCHITECTURES is empty for target
     68        "directed_tests.kernel.hipPrintfKernel".
     69      
     70      
  >> 71      CMake Error in CMakeLists.txt:
     72        HIP_ARCHITECTURES is empty for target
     73        "directed_tests.kernel.hipTestMemKernel".
     74      
     75      
  >> 76      CMake Error in CMakeLists.txt:
     77        HIP_ARCHITECTURES is empty for target
     78        "directed_tests.kernel.hipTestConstant".
     79      
     80      
  >> 81      CMake Error in CMakeLists.txt:
     82        HIP_ARCHITECTURES is empty for target
     83        "directed_tests.kernel.hipDynamicShared".
     84      
     85      
  >> 86      CMake Error in CMakeLists.txt:
     87        HIP_ARCHITECTURES is empty for target
     88        "directed_tests.kernel.hipDynamicShared2".
     89      
     90      
  >> 91      CMake Error in CMakeLists.txt:
     92        HIP_ARCHITECTURES is empty for target
     93        "directed_tests.kernel.hipEmptyKernel".
     94      
     95      
  >> 96      CMake Error in CMakeLists.txt:
     97        HIP_ARCHITECTURES is empty for target
     98        "directed_tests.kernel.hipExtLaunchKernelGGL".
     99      
     100     
  >> 101     CMake Error in CMakeLists.txt:
     102       HIP_ARCHITECTURES is empty for target
     103       "directed_tests.kernel.hipGridLaunch".
     104     
     105     
  >> 106     CMake Error in CMakeLists.txt:
     107       HIP_ARCHITECTURES is empty for target
     108       "directed_tests.kernel.hipLanguageExtensions".
     109     
     110     
  >> 111     CMake Error in CMakeLists.txt:
     112       HIP_ARCHITECTURES is empty for target
     113       "directed_tests.kernel.hipLaunchParmFunctor".
     114     
     115     
  >> 116     CMake Error in CMakeLists.txt:
     117       HIP_ARCHITECTURES is empty for target "directed_tests.kernel.hipShflTests".
     118     
     119     
  >> 120     CMake Error in CMakeLists.txt:
     121       HIP_ARCHITECTURES is empty for target
     122       "directed_tests.kernel.hipShflUpDownTest".
     123     
     124     
  >> 125     CMake Error in CMakeLists.txt:
     126       HIP_ARCHITECTURES is empty for target
     127       "directed_tests.kernel.hipTestGlobalVariable".
     128     
     129     
  >> 130     CMake Error in CMakeLists.txt:
     131       HIP_ARCHITECTURES is empty for target
     132       "directed_tests.kernel.inline_asm_vadd".
     133     
     134     
  >> 135     CMake Error in CMakeLists.txt:
     136       HIP_ARCHITECTURES is empty for target
     137       "directed_tests.printf.hipPrintfWidthPrecision".
     138     
     139     
  >> 140     CMake Error in CMakeLists.txt:
     141       HIP_ARCHITECTURES is empty for target
     142       "directed_tests.printf.hipPrintfFlags".
     143     
     144     
  >> 145     CMake Error in CMakeLists.txt:
     146       HIP_ARCHITECTURES is empty for target
     147       "directed_tests.printf.hipPrintfAltForms".
     148     
     149     
  >> 150     CMake Error in CMakeLists.txt:
     151       HIP_ARCHITECTURES is empty for target
     152       "directed_tests.printf.hipPrintfBasic".
     153     
     154     
  >> 155     CMake Error in CMakeLists.txt:
     156       HIP_ARCHITECTURES is empty for target
     157       "directed_tests.printf.hipPrintfManyDevices".
     158     
     159     
  >> 160     CMake Error in CMakeLists.txt:
     161       HIP_ARCHITECTURES is empty for target
     162       "directed_tests.printf.hipPrintfSpecifiers".
     163     
     164     
  >> 165     CMake Error in CMakeLists.txt:
     166       HIP_ARCHITECTURES is empty for target
     167       "directed_tests.printf.hipPrintfManyWaves".
     168     
     169     
  >> 170     CMake Error in CMakeLists.txt:
     171       HIP_ARCHITECTURES is empty for target
     172       "directed_tests.printf.hipPrintfStar".
     173     
     174     
  >> 175     CMake Error in CMakeLists.txt:
     176       HIP_ARCHITECTURES is empty for target
     177       "directed_tests.cppstd.hipInvocable14".
     178     
     179     
  >> 180     CMake Error in CMakeLists.txt:
     181       HIP_ARCHITECTURES is empty for target
     182       "directed_tests.cppstd.hipInvocable11".
     183     
     184     
  >> 185     CMake Error in CMakeLists.txt:
     186       HIP_ARCHITECTURES is empty for target
     187       "directed_tests.cppstd.hipInvocable17".
     188     
     189     
  >> 190     CMake Error in CMakeLists.txt:
     191       HIP_ARCHITECTURES is empty for target
     192       "directed_tests.cppstd.hipInvocable11Fallback".
     193     
     194     
  >> 195     CMake Error in CMakeLists.txt:
     196       HIP_ARCHITECTURES is empty for target
     197       "directed_tests.Negative.memory.hipMemcpyToSymbolAsync".
     198     
     199     
  >> 200     CMake Error in CMakeLists.txt:
     201       HIP_ARCHITECTURES is empty for target
     202       "directed_tests.Negative.memory.hipMemcpyFromSymbol".
     203     
     204     
  >> 205     CMake Error in CMakeLists.txt:
     206       HIP_ARCHITECTURES is empty for target
     207       "directed_tests.Negative.memory.hipMemcpyFromSymbolAsync".
     208     
     209     
  >> 210     CMake Error in CMakeLists.txt:
     211       HIP_ARCHITECTURES is empty for target
     212       "directed_tests.Negative.memory.hipMemcpyToSymbol".
     213     
     214     
  >> 215     CMake Error in CMakeLists.txt:
     216       HIP_ARCHITECTURES is empty for target "directed_tests.context.hipMemsetD8".
     217     
     218     
  >> 219     CMake Error in CMakeLists.txt:
     220       HIP_ARCHITECTURES is empty for target
     221       "directed_tests.context.hipCtx_simple".
     222     
     223     
  >> 224     CMake Error in CMakeLists.txt:
     225       HIP_ARCHITECTURES is empty for target
     226       "directed_tests.compiler.hipClassKernel".
     227     
     228     
  >> 229     CMake Error in CMakeLists.txt:
     230       HIP_ARCHITECTURES is empty for target "directed_tests.hipEnvVar".
     231     
     232     
  >> 233     CMake Error in CMakeLists.txt:
     234       HIP_ARCHITECTURES is empty for target "directed_tests.hipEnvVarDriver".
     235     
     236     
  >> 237     CMake Error in CMakeLists.txt:
     238       HIP_ARCHITECTURES is empty for target "directed_tests.hipHcc".
     239     
     240     
  >> 241     CMake Error in CMakeLists.txt:
     242       HIP_ARCHITECTURES is empty for target
     243       "directed_tests.Functional.host.hipFloat16".
     244     
     245     
  >> 246     CMake Error in CMakeLists.txt:
     247       HIP_ARCHITECTURES is empty for target
     248       "directed_tests.Functional.device.hipFuncSetDevice".
     249     
     250     
  >> 251     CMake Error in CMakeLists.txt:
     252       HIP_ARCHITECTURES is empty for target
     253       "directed_tests.Functional.device.hipFuncSetDeviceFlags".
     254     
     255     
  >> 256     CMake Error in CMakeLists.txt:
     257       HIP_ARCHITECTURES is empty for target
     258       "directed_tests.Functional.device.hipFuncGetDevice".
     259     
     260     
  >> 261     CMake Error in CMakeLists.txt:
     262       HIP_ARCHITECTURES is empty for target
     263       "directed_tests.Functional.device.hipFuncDeviceSynchronize".
     264     
     265     
  >> 266     CMake Error in CMakeLists.txt:
     267       HIP_ARCHITECTURES is empty for target
     268       "directed_tests.hostcall.hipHostcallPrintThings".
     269     
     270     
  >> 271     CMake Error in CMakeLists.txt:
     272       HIP_ARCHITECTURES is empty for target
     273       "directed_tests.hostcall.hipHostcallFuncCall".
     274     
     275     
  >> 276     CMake Error in CMakeLists.txt:
     277       HIP_ARCHITECTURES is empty for target
     278       "directed_tests.deviceLib.hipFloatMathPrecise".
     279     
     280     
  >> 281     CMake Error in CMakeLists.txt:
     282       HIP_ARCHITECTURES is empty for target
     283       "directed_tests.deviceLib.hip_threadfence_system".
     284     
     285     
  >> 286     CMake Error in CMakeLists.txt:
     287       HIP_ARCHITECTURES is empty for target
     288       "directed_tests.deviceLib.hipTestDevice".
     289     
     290     
  >> 291     CMake Error in CMakeLists.txt:
     292       HIP_ARCHITECTURES is empty for target
     293       "directed_tests.deviceLib.hip_test_ldg".
     294     
     295     
  >> 296     CMake Error in CMakeLists.txt:
     297       HIP_ARCHITECTURES is empty for target "directed_tests.deviceLib.hip_popc".
     298     
     299     
  >> 300     CMake Error in CMakeLists.txt:
     301       HIP_ARCHITECTURES is empty for target
     302       "directed_tests.deviceLib.hip_bitinsert".
     303     
     304     
  >> 305     CMake Error in CMakeLists.txt:
     306       HIP_ARCHITECTURES is empty for target
     307       "directed_tests.deviceLib.hip_ballot".
     308     
     309     
  >> 310     CMake Error in CMakeLists.txt:
     311       HIP_ARCHITECTURES is empty for target
     312       "directed_tests.deviceLib.hipTestIncludeMath".
     313     
     314     
  >> 315     CMake Error in CMakeLists.txt:
     316       HIP_ARCHITECTURES is empty for target
     317       "directed_tests.deviceLib.hipThreadFence".
     318     
     319     
  >> 320     CMake Error in CMakeLists.txt:
     321       HIP_ARCHITECTURES is empty for target
     322       "directed_tests.deviceLib.hipTestHalf".
     323     
     324     
  >> 325     CMake Error in CMakeLists.txt:
     326       HIP_ARCHITECTURES is empty for target
     327       "directed_tests.deviceLib.hipTestClock".
     328     
     329     
  >> 330     CMake Error in CMakeLists.txt:
     331       HIP_ARCHITECTURES is empty for target
     332       "directed_tests.deviceLib.hipTestAtomicAdd".
     333     
     334     
  >> 335     CMake Error in CMakeLists.txt:
     336       HIP_ARCHITECTURES is empty for target
     337       "directed_tests.deviceLib.hipSinglePrecisionMathDevice".
     338     
     339     
  >> 340     CMake Error in CMakeLists.txt:
     341       HIP_ARCHITECTURES is empty for target
     342       "directed_tests.deviceLib.hipIntegerIntrinsics".
     343     
     344     
  >> 345     CMake Error in CMakeLists.txt:
     346       HIP_ARCHITECTURES is empty for target
     347       "directed_tests.deviceLib.hipSinglePrecisionIntrinsics".
     348     
     349     
  >> 350     CMake Error in CMakeLists.txt:
     351       HIP_ARCHITECTURES is empty for target
     352       "directed_tests.deviceLib.hipMathFunctions".
     353     
     354     
  >> 355     CMake Error in CMakeLists.txt:
     356       HIP_ARCHITECTURES is empty for target
     357       "directed_tests.deviceLib.hipHalf2Comparision".
     358     
     359     
  >> 360     CMake Error in CMakeLists.txt:
     361       HIP_ARCHITECTURES is empty for target
     362       "directed_tests.deviceLib.hipDoublePrecisionIntrinsics".
     363     
     364     
  >> 365     CMake Error in CMakeLists.txt:
     366       HIP_ARCHITECTURES is empty for target
     367       "directed_tests.deviceLib.hipFloatMath".
     368     
     369     
  >> 370     CMake Error in CMakeLists.txt:
     371       HIP_ARCHITECTURES is empty for target
     372       "directed_tests.deviceLib.hipBfloat16".
     373     
     374     
  >> 375     CMake Error in CMakeLists.txt:
     376       HIP_ARCHITECTURES is empty for target
     377       "directed_tests.deviceLib.hipTestDotFunctions".
     378     
     379     
  >> 380     CMake Error in CMakeLists.txt:
     381       HIP_ARCHITECTURES is empty for target
     382       "directed_tests.deviceLib.hip_anyall".
     383     
     384     
  >> 385     CMake Error in CMakeLists.txt:
     386       HIP_ARCHITECTURES is empty for target
     387       "directed_tests.deviceLib.hipTestDeviceSymbol".
     388     
     389     
  >> 390     CMake Error in CMakeLists.txt:
     391       HIP_ARCHITECTURES is empty for target
     392       "directed_tests.deviceLib.hip_test_syncthreads_and".
     393     
     394     
  >> 395     CMake Error in CMakeLists.txt:
     396       HIP_ARCHITECTURES is empty for target "directed_tests.deviceLib.hip_clz".
     397     
     398     
  >> 399     CMake Error in CMakeLists.txt:
     400       HIP_ARCHITECTURES is empty for target
     401       "directed_tests.deviceLib.hipVectorTypesDevice".
     402     
     403     
  >> 404     CMake Error in CMakeLists.txt:
     405       HIP_ARCHITECTURES is empty for target
     406       "directed_tests.deviceLib.hipTestFMA".
     407     
     408     
  >> 409     CMake Error in CMakeLists.txt:
     410       HIP_ARCHITECTURES is empty for target
     411       "directed_tests.deviceLib.hipDeviceMemcpy".
     412     
     413     
  >> 414     CMake Error in CMakeLists.txt:
     415       HIP_ARCHITECTURES is empty for target "directed_tests.deviceLib.hip_ffs".
     416     
     417     
  >> 418     CMake Error in CMakeLists.txt:
     419       HIP_ARCHITECTURES is empty for target
     420       "directed_tests.deviceLib.hipTestNew".
     421     
     422     
  >> 423     CMake Error in CMakeLists.txt:
     424       HIP_ARCHITECTURES is empty for target
     425       "directed_tests.deviceLib.hipVersion".
     426     
     427     
  >> 428     CMake Error in CMakeLists.txt:
     429       HIP_ARCHITECTURES is empty for target "directed_tests.deviceLib.hip_mbcnt".
     430     
     431     
  >> 432     CMake Error in CMakeLists.txt:
     433       HIP_ARCHITECTURES is empty for target
     434       "directed_tests.deviceLib.hipTestNativeHalf".
     435     
     436     
  >> 437     CMake Error in CMakeLists.txt:
     438       HIP_ARCHITECTURES is empty for target
     439       "directed_tests.deviceLib.hipSinglePrecisionMathHost".
     440     
     441     
  >> 442     CMake Error in CMakeLists.txt:
     443       HIP_ARCHITECTURES is empty for target
     444       "directed_tests.deviceLib.hip_floatnTM".
     445     
     446     
  >> 447     CMake Error in CMakeLists.txt:
     448       HIP_ARCHITECTURES is empty for target "directed_tests.deviceLib.hip_trig".
     449     
     450     
  >> 451     CMake Error in CMakeLists.txt:
     452       HIP_ARCHITECTURES is empty for target
     453       "directed_tests.deviceLib.hipDoublePrecisionMathHost".
     454     
     455     
  >> 456     CMake Error in CMakeLists.txt:
     457       HIP_ARCHITECTURES is empty for target "directed_tests.deviceLib.hip_brev".
     458     
     459     
  >> 460     CMake Error in CMakeLists.txt:
     461       HIP_ARCHITECTURES is empty for target
     462       "directed_tests.deviceLib.hipLaunchKernelFunc".
     463     
     464     
  >> 465     CMake Error in CMakeLists.txt:
     466       HIP_ARCHITECTURES is empty for target
     467       "directed_tests.deviceLib.hipVectorTypes".
     468     
     469     
  >> 470     CMake Error in CMakeLists.txt:
     471       HIP_ARCHITECTURES is empty for target
     472       "directed_tests.deviceLib.hip_test_syncthreads_or".
     473     
     474     
  >> 475     CMake Error in CMakeLists.txt:
     476       HIP_ARCHITECTURES is empty for target
     477       "directed_tests.deviceLib.hipDoublePrecisionMathDevice".
     478     
     479     
  >> 480     CMake Error in CMakeLists.txt:
     481       HIP_ARCHITECTURES is empty for target
     482       "directed_tests.deviceLib.hipStdComplex".
     483     
     484     
  >> 485     CMake Error in CMakeLists.txt:
     486       HIP_ARCHITECTURES is empty for target
     487       "directed_tests.deviceLib.hipSimpleAtomicsTest".
     488     
     489     
  >> 490     CMake Error in CMakeLists.txt:
     491       HIP_ARCHITECTURES is empty for target
     492       "directed_tests.deviceLib.hip_test_syncthreads_count".
     493     
     494     
  >> 495     CMake Error in CMakeLists.txt:
     496       HIP_ARCHITECTURES is empty for target
     497       "directed_tests.deviceLib.hip_bitextract".
     498     
     499     
  >> 500     CMake Error in CMakeLists.txt:
     501       HIP_ARCHITECTURES is empty for target
     502       "directed_tests.deviceLib.hipComplex".
     503     
     504     
  >> 505     CMake Error in CMakeLists.txt:
     506       HIP_ARCHITECTURES is empty for target
     507       "directed_tests.ipc.hipMultiProcIpcEvent".
     508     
     509     
  >> 510     CMake Error in CMakeLists.txt:
     511       HIP_ARCHITECTURES is empty for target
     512       "directed_tests.ipc.hipMultiProcIpcMem".
     513     
     514     
  >> 515     CMake Error in CMakeLists.txt:
     516       HIP_ARCHITECTURES is empty for target
     517       "directed_tests.texture.hipTextureObj2D".
     518     
     519     
  >> 520     CMake Error in CMakeLists.txt:
     521       HIP_ARCHITECTURES is empty for target
     522       "directed_tests.texture.hipTextureMipmapObj2D".
     523     
     524     
  >> 525     CMake Error in CMakeLists.txt:
     526       HIP_ARCHITECTURES is empty for target
     527       "directed_tests.texture.hipTexObjPitch".
     528     
     529     
  >> 530     CMake Error in CMakeLists.txt:
     531       HIP_ARCHITECTURES is empty for target
     532       "directed_tests.texture.hipGetChanDesc".
     533     
     534     
  >> 535     CMake Error in CMakeLists.txt:
     536       HIP_ARCHITECTURES is empty for target
     537       "directed_tests.p2p.hipPeerToPeer_simple".
     538     
     539     
  >> 540     CMake Error in CMakeLists.txt:
     541       HIP_ARCHITECTURES is empty for target "directed_tests.hiprtc.saxpy_pch".
     542     
     543     
  >> 544     CMake Error in CMakeLists.txt:
     545       HIP_ARCHITECTURES is empty for target
     546       "directed_tests.hiprtc.hiprtcGetLoweredName".
     547     
     548     
  >> 549     CMake Error in CMakeLists.txt:
     550       HIP_ARCHITECTURES is empty for target "directed_tests.hiprtc.saxpy".
     551     
     552     
  >> 553     CMake Error in CMakeLists.txt:
     554       HIP_ARCHITECTURES is empty for target
     555       "directed_tests.runtimeApi.streamOperations.hipstream_operations".
     556     
     557     
  >> 558     CMake Error in CMakeLists.txt:
     559       HIP_ARCHITECTURES is empty for target
     560       "directed_tests.runtimeApi.p2p.hipDeviceGetP2PAttribute".
     561     
     562     
  >> 563     CMake Error in CMakeLists.txt:
     564       HIP_ARCHITECTURES is empty for target
     565       "directed_tests.runtimeApi.multiThread.hipMultiThreadStreams1".
     566     
     567     
  >> 568     CMake Error in CMakeLists.txt:
     569       HIP_ARCHITECTURES is empty for target
     570       "directed_tests.runtimeApi.multiThread.hipMultiThreadDevice".
     571     
     572     
  >> 573     CMake Error in CMakeLists.txt:
     574       HIP_ARCHITECTURES is empty for target
     575       "directed_tests.runtimeApi.multiThread.hipMultiThreadStreams2".
     576     
     577     
  >> 578     CMake Error in CMakeLists.txt:
     579       HIP_ARCHITECTURES is empty for target
     580       "directed_tests.runtimeApi.memory.hipTestMemcpyPin".
     581     
     582     
  >> 583     CMake Error in CMakeLists.txt:
     584       HIP_ARCHITECTURES is empty for target
     585       "directed_tests.runtimeApi.memory.hipRandomMemcpyAsync".
     586     
     587     
  >> 588     CMake Error in CMakeLists.txt:
     589       HIP_ARCHITECTURES is empty for target
     590       "directed_tests.runtimeApi.memory.hipPointerAttributes".
     591     
     592     
  >> 593     CMake Error in CMakeLists.txt:
     594       HIP_ARCHITECTURES is empty for target
     595       "directed_tests.runtimeApi.memory.hipMemcpyPeerAsync".
     596     
     597     
  >> 598     CMake Error in CMakeLists.txt:
     599       HIP_ARCHITECTURES is empty for target
     600       "directed_tests.runtimeApi.memory.hipMemcpyAsync2".
     601     
     602     
  >> 603     CMake Error in CMakeLists.txt:
     604       HIP_ARCHITECTURES is empty for target
     605       "directed_tests.runtimeApi.memory.hipMemcpyDtoDAsync".
     606     
     607     
  >> 608     CMake Error in CMakeLists.txt:
     609       HIP_ARCHITECTURES is empty for target
     610       "directed_tests.runtimeApi.memory.hipMemsetInvalidPtr".
     611     
     612     
  >> 613     CMake Error in CMakeLists.txt:
     614       HIP_ARCHITECTURES is empty for target
     615       "directed_tests.runtimeApi.memory.hipMemcpyNegativeMThrdMSize".
     616     
     617     
  >> 618     CMake Error in CMakeLists.txt:
     619       HIP_ARCHITECTURES is empty for target
     620       "directed_tests.runtimeApi.memory.p2p_copy_coherency".
     621     
     622     
  >> 623     CMake Error in CMakeLists.txt:
     624       HIP_ARCHITECTURES is empty for target
     625       "directed_tests.runtimeApi.memory.d2dMemCpyWithPinnedHostMemory".
     626     
     627     
  >> 628     CMake Error in CMakeLists.txt:
     629       HIP_ARCHITECTURES is empty for target
     630       "directed_tests.runtimeApi.memory.hipMemcpy2DAsync".
     631     
     632     
  >> 633     CMake Error in CMakeLists.txt:
     634       HIP_ARCHITECTURES is empty for target
     635       "directed_tests.runtimeApi.memory.hipMemcpyDtoD".
     636     
     637     
  >> 638     CMake Error in CMakeLists.txt:
     639       HIP_ARCHITECTURES is empty for target
     640       "directed_tests.runtimeApi.memory.hipHostGetFlags".
     641     
     642     
  >> 643     CMake Error in CMakeLists.txt:
     644       HIP_ARCHITECTURES is empty for target
     645       "directed_tests.runtimeApi.memory.hipMemcpyPeer".
     646     
     647     
  >> 648     CMake Error in CMakeLists.txt:
     649       HIP_ARCHITECTURES is empty for target
     650       "directed_tests.runtimeApi.memory.hipMemset3DRegressMultiThread".
     651     
     652     
  >> 653     CMake Error in CMakeLists.txt:
     654       HIP_ARCHITECTURES is empty for target
     655       "directed_tests.runtimeApi.memory.hipMemset2DAsyncMultiThreadAndKernel".
     656     
     657     
  >> 658     CMake Error in CMakeLists.txt:
     659       HIP_ARCHITECTURES is empty for target
     660       "directed_tests.runtimeApi.memory.hipMemcpy_simple".
     661     
     662     
  >> 663     CMake Error in CMakeLists.txt:
     664       HIP_ARCHITECTURES is empty for target
     665       "directed_tests.runtimeApi.memory.hipMemset".
     666     
     667     
  >> 668     CMake Error in CMakeLists.txt:
     669       HIP_ARCHITECTURES is empty for target
     670       "directed_tests.runtimeApi.memory.hipMallocManaged".
     671     
     672     
  >> 673     CMake Error in CMakeLists.txt:
     674       HIP_ARCHITECTURES is empty for target
     675       "directed_tests.runtimeApi.memory.hipMemset3DFunctional".
     676     
     677     
  >> 678     CMake Error in CMakeLists.txt:
     679       HIP_ARCHITECTURES is empty for target
     680       "directed_tests.runtimeApi.memory.hipMemoryAllocateCoherent".
     681     
     682     
  >> 683     CMake Error in CMakeLists.txt:
     684       HIP_ARCHITECTURES is empty for target
     685       "directed_tests.runtimeApi.memory.hipHostRegister".
     686     
     687     
  >> 688     CMake Error in CMakeLists.txt:
     689       HIP_ARCHITECTURES is empty for target
     690       "directed_tests.runtimeApi.memory.hipMemPtrGetInfo".
     691     
     692     
  >> 693     CMake Error in CMakeLists.txt:
     694       HIP_ARCHITECTURES is empty for target
     695       "directed_tests.runtimeApi.memory.hipMemcpyAtoH".
     696     
     697     
  >> 698     CMake Error in CMakeLists.txt:
     699       HIP_ARCHITECTURES is empty for target
     700       "directed_tests.runtimeApi.memory.hipMultiMemcpyMultiThread".
     701     
     702     
  >> 703     CMake Error in CMakeLists.txt:
     704       HIP_ARCHITECTURES is empty for target
     705       "directed_tests.runtimeApi.memory.hipMemsetAsyncMultiThread".
     706     
     707     
  >> 708     CMake Error in CMakeLists.txt:
     709       HIP_ARCHITECTURES is empty for target
     710       "directed_tests.runtimeApi.memory.hipMallocManaged_MultiScenario".
     711     
     712     
  >> 713     CMake Error in CMakeLists.txt:
     714       HIP_ARCHITECTURES is empty for target
     715       "directed_tests.runtimeApi.memory.hipMemset2D".
     716     
     717     
  >> 718     CMake Error in CMakeLists.txt:
     719       HIP_ARCHITECTURES is empty for target
     720       "directed_tests.runtimeApi.memory.hipMemcpy".
     721     
     722     
  >> 723     CMake Error in CMakeLists.txt:
     724       HIP_ARCHITECTURES is empty for target
     725       "directed_tests.runtimeApi.memory.hipMultiMemcpyMultiThrdMultiStrm".
     726     
     727     
  >> 728     CMake Error in CMakeLists.txt:
     729       HIP_ARCHITECTURES is empty for target
     730       "directed_tests.runtimeApi.memory.hipIpcMemAccessTest".
     731     
     732     
  >> 733     CMake Error in CMakeLists.txt:
     734       HIP_ARCHITECTURES is empty for target
     735       "directed_tests.runtimeApi.memory.hipMemcpy3D".
     736     
     737     
  >> 738     CMake Error in CMakeLists.txt:
     739       HIP_ARCHITECTURES is empty for target
     740       "directed_tests.runtimeApi.memory.hipHostMalloc".
     741     
     742     
  >> 743     CMake Error in CMakeLists.txt:
     744       HIP_ARCHITECTURES is empty for target
     745       "directed_tests.runtimeApi.memory.hipMallocConcurrency".
     746     
     747     
  >> 748     CMake Error in CMakeLists.txt:
     749       HIP_ARCHITECTURES is empty for target
     750       "directed_tests.runtimeApi.memory.hipMemcpy2D".
     751     
     752     
  >> 753     CMake Error in CMakeLists.txt:
     754       HIP_ARCHITECTURES is empty for target
     755       "directed_tests.runtimeApi.memory.hipMemset3DNegative".
     756     
     757     
  >> 758     CMake Error in CMakeLists.txt:
     759       HIP_ARCHITECTURES is empty for target
     760       "directed_tests.runtimeApi.memory.hipMemcpy2DFromArray".
     761     
     762     
  >> 763     CMake Error in CMakeLists.txt:
     764       HIP_ARCHITECTURES is empty for target
     765       "directed_tests.runtimeApi.memory.hipMemcpy2DFromArrayAsync".
     766     
     767     
  >> 768     CMake Error in CMakeLists.txt:
     769       HIP_ARCHITECTURES is empty for target
     770       "directed_tests.runtimeApi.memory.hipMemcpy2D_simple".
     771     
     772     
  >> 773     CMake Error in CMakeLists.txt:
     774       HIP_ARCHITECTURES is empty for target
     775       "directed_tests.runtimeApi.memory.hipMemcpyAll".
     776     
     777     
  >> 778     CMake Error in CMakeLists.txt:
     779       HIP_ARCHITECTURES is empty for target
     780       "directed_tests.runtimeApi.memory.hipMemcpyHtoA".
     781     
     782     
  >> 783     CMake Error in CMakeLists.txt:
     784       HIP_ARCHITECTURES is empty for target
     785       "directed_tests.runtimeApi.memory.hipMemcpyWithStream".
     786     
     787     
  >> 788     CMake Error in CMakeLists.txt:
     789       HIP_ARCHITECTURES is empty for target
     790       "directed_tests.runtimeApi.memory.hipMemcpyWithStreamMultiThread".
     791     
     792     
  >> 793     CMake Error in CMakeLists.txt:
     794       HIP_ARCHITECTURES is empty for target
     795       "directed_tests.runtimeApi.memory.hipMemoryAllocateCoherentDriver".
     796     
     797     
  >> 798     CMake Error in CMakeLists.txt:
     799       HIP_ARCHITECTURES is empty for target
     800       "directed_tests.runtimeApi.memory.hipMemset3D".
     801     
     802     
  >> 803     CMake Error in CMakeLists.txt:
     804       HIP_ARCHITECTURES is empty for target
     805       "directed_tests.runtimeApi.memory.hipMemsetAsyncAndKernel".
     806     
     807     
  >> 808     CMake Error in CMakeLists.txt:
     809       HIP_ARCHITECTURES is empty for target
     810       "directed_tests.runtimeApi.module.hipModuleOccupancyMaxPotentialActiveBlockSize".
     811     
     812     
  >> 813     CMake Error in CMakeLists.txt:
     814       HIP_ARCHITECTURES is empty for target
     815       "directed_tests.runtimeApi.module.hipModuleNegative".
     816     
     817     
  >> 818     CMake Error in CMakeLists.txt:
     819       HIP_ARCHITECTURES is empty for target
     820       "directed_tests.runtimeApi.module.hipModuleLoadUnloadStress".
     821     
     822     
  >> 823     CMake Error in CMakeLists.txt:
     824       HIP_ARCHITECTURES is empty for target
     825       "directed_tests.runtimeApi.module.hipModuleLoadMultiThreaded".
     826     
     827     
  >> 828     CMake Error in CMakeLists.txt:
     829       HIP_ARCHITECTURES is empty for target
     830       "directed_tests.runtimeApi.module.hipModuleLoadMultProcessOnMultGPU".
     831     
     832     
  >> 833     CMake Error in CMakeLists.txt:
     834       HIP_ARCHITECTURES is empty for target
     835       "directed_tests.runtimeApi.module.hipModuleLoadDataMultThreadOnMultGPU".
     836     
     837     
  >> 838     CMake Error in CMakeLists.txt:
     839       HIP_ARCHITECTURES is empty for target
     840       "directed_tests.runtimeApi.module.hipModuleLoadData".
     841     
     842     
  >> 843     CMake Error in CMakeLists.txt:
     844       HIP_ARCHITECTURES is empty for target
     845       "directed_tests.runtimeApi.module.hipModuleLaunchKernel".
     846     
     847     
  >> 848     CMake Error in CMakeLists.txt:
     849       HIP_ARCHITECTURES is empty for target
     850       "directed_tests.runtimeApi.module.hipModule".
     851     
     852     
  >> 853     CMake Error in CMakeLists.txt:
     854       HIP_ARCHITECTURES is empty for target
     855       "directed_tests.runtimeApi.module.hipFuncSetSharedMemConfig".
     856     
     857     
  >> 858     CMake Error in CMakeLists.txt:
     859       HIP_ARCHITECTURES is empty for target
     860       "directed_tests.runtimeApi.module.hipFuncSetAttribute".
     861     
     862     
  >> 863     CMake Error in CMakeLists.txt:
     864       HIP_ARCHITECTURES is empty for target
     865       "directed_tests.runtimeApi.module.hipFuncGetAttributes".
     866     
     867     
  >> 868     CMake Error in CMakeLists.txt:
     869       HIP_ARCHITECTURES is empty for target
     870       "directed_tests.runtimeApi.module.hipOpenCLCOTest".
     871     
     872     
  >> 873     CMake Error in CMakeLists.txt:
     874       HIP_ARCHITECTURES is empty for target
     875       "directed_tests.runtimeApi.module.hipExtLaunchMultiKernelMultiDevice".
     876     
     877     
  >> 878     CMake Error in CMakeLists.txt:
     879       HIP_ARCHITECTURES is empty for target
     880       "directed_tests.runtimeApi.module.hipFuncSetCacheConfig".
     881     
     882     
  >> 883     CMake Error in CMakeLists.txt:
     884       HIP_ARCHITECTURES is empty for target
     885       "directed_tests.runtimeApi.module.hipModuleTexture2dDrv".
     886     
     887     
  >> 888     CMake Error in CMakeLists.txt:
     889       HIP_ARCHITECTURES is empty for target
     890       "directed_tests.runtimeApi.module.hipExtModuleLaunchKernel".
     891     
     892     
  >> 893     CMake Error in CMakeLists.txt:
     894       HIP_ARCHITECTURES is empty for target
     895       "directed_tests.runtimeApi.module.hipExtLaunchKernelGGL".
     896     
     897     
  >> 898     CMake Error in CMakeLists.txt:
     899       HIP_ARCHITECTURES is empty for target
     900       "directed_tests.runtimeApi.module.hipModuleGetGlobal".
     901     
     902     
  >> 903     CMake Error in CMakeLists.txt:
     904       HIP_ARCHITECTURES is empty for target
     905       "directed_tests.runtimeApi.synchronization.copy_coherency".
     906     
     907     
  >> 908     CMake Error in CMakeLists.txt:
     909       HIP_ARCHITECTURES is empty for target
     910       "directed_tests.runtimeApi.synchronization.cache_coherency_cpu_gpu".
     911     
     912     
  >> 913     CMake Error in CMakeLists.txt:
     914       HIP_ARCHITECTURES is empty for target
     915       "directed_tests.runtimeApi.synchronization.cache_coherency_gpu_gpu".
     916     
     917     
  >> 918     CMake Error in CMakeLists.txt:
     919       HIP_ARCHITECTURES is empty for target
     920       "directed_tests.runtimeApi.stream.hipStreamL5".
     921     
     922     
  >> 923     CMake Error in CMakeLists.txt:
     924       HIP_ARCHITECTURES is empty for target
     925       "directed_tests.runtimeApi.stream.hipStreamGetPriority".
     926     
     927     
  >> 928     CMake Error in CMakeLists.txt:
     929       HIP_ARCHITECTURES is empty for target
     930       "directed_tests.runtimeApi.stream.hipStreamGetCUMask".
     931     
     932     
  >> 933     CMake Error in CMakeLists.txt:
     934       HIP_ARCHITECTURES is empty for target
     935       "directed_tests.runtimeApi.stream.hipStreamACb_ThrdBehaviour".
     936     
     937     
  >> 938     CMake Error in CMakeLists.txt:
     939       HIP_ARCHITECTURES is empty for target
     940       "directed_tests.runtimeApi.stream.hipStreamACb_MStrm_Mgpu".
     941     
     942     
  >> 943     CMake Error in CMakeLists.txt:
     944       HIP_ARCHITECTURES is empty for target
     945       "directed_tests.runtimeApi.stream.hipNullStream".
     946     
     947     
  >> 948     CMake Error in CMakeLists.txt:
     949       HIP_ARCHITECTURES is empty for target
     950       "directed_tests.runtimeApi.stream.StreamAddCallback".
     951     
     952     
  >> 953     CMake Error in CMakeLists.txt:
     954       HIP_ARCHITECTURES is empty for target
     955       "directed_tests.runtimeApi.stream.hipStreamACb_StrmSyncTiming".
     956     
     957     
  >> 958     CMake Error in CMakeLists.txt:
     959       HIP_ARCHITECTURES is empty for target
     960       "directed_tests.runtimeApi.stream.hipStreamAddCallback".
     961     
     962     
  >> 963     CMake Error in CMakeLists.txt:
     964       HIP_ARCHITECTURES is empty for target
     965       "directed_tests.runtimeApi.stream.hipStreamAddCallbackCatch".
     966     
     967     
  >> 968     CMake Error in CMakeLists.txt:
     969       HIP_ARCHITECTURES is empty for target
     970       "directed_tests.runtimeApi.stream.hipStreamACb_MultiCalls".
     971     
     972     
  >> 973     CMake Error in CMakeLists.txt:
     974       HIP_ARCHITECTURES is empty for target
     975       "directed_tests.runtimeApi.stream.hipStreamACb_order".
     976     
     977     
  >> 978     CMake Error in CMakeLists.txt:
     979       HIP_ARCHITECTURES is empty for target
     980       "directed_tests.runtimeApi.stream.hipAPIStreamDisable".
     981     
     982     
  >> 983     CMake Error in CMakeLists.txt:
     984       HIP_ARCHITECTURES is empty for target
     985       "directed_tests.runtimeApi.stream.hipMultiStreams".
     986     
     987     
  >> 988     CMake Error in CMakeLists.txt:
     989       HIP_ARCHITECTURES is empty for target
     990       "directed_tests.runtimeApi.stream.hipStreamGetFlags".
     991     
     992     
  >> 993     CMake Error in CMakeLists.txt:
     994       HIP_ARCHITECTURES is empty for target
     995       "directed_tests.runtimeApi.stream.hipStreamWithCUMask".
     996     
     997     
  >> 998     CMake Error in CMakeLists.txt:
     999       HIP_ARCHITECTURES is empty for target
     1000      "directed_tests.runtimeApi.stream.hipStreamACb_AltEnqueue".
     1001    
     1002    
  >> 1003    CMake Error in CMakeLists.txt:
     1004      HIP_ARCHITECTURES is empty for target
     1005      "directed_tests.runtimeApi.stream.hipAPIStreamEnable".
     1006    
     1007    
  >> 1008    CMake Error in CMakeLists.txt:
     1009      HIP_ARCHITECTURES is empty for target
     1010      "directed_tests.runtimeApi.stream.hipStreamSync2".
     1011    
     1012    
  >> 1013    CMake Error in CMakeLists.txt:
     1014      HIP_ARCHITECTURES is empty for target
     1015      "directed_tests.runtimeApi.graph.hipGraph".
     1016    
     1017    
  >> 1018    CMake Error in CMakeLists.txt:
     1019      HIP_ARCHITECTURES is empty for target
     1020      "directed_tests.runtimeApi.cooperativeGrps.simple_grid_group_barrier".
     1021    
     1022    
  >> 1023    CMake Error in CMakeLists.txt:
     1024      HIP_ARCHITECTURES is empty for target
     1025      "directed_tests.runtimeApi.cooperativeGrps.cooperative_streams".
     1026    
     1027    
  >> 1028    CMake Error in CMakeLists.txt:
     1029      HIP_ARCHITECTURES is empty for target
     1030      "directed_tests.runtimeApi.cooperativeGrps.multi_gpu_api_failure_tests".
     1031    
     1032    
  >> 1033    CMake Error in CMakeLists.txt:
     1034      HIP_ARCHITECTURES is empty for target
     1035      "directed_tests.runtimeApi.cooperativeGrps.hipCGMultiGridGroupType".
     1036    
     1037    
  >> 1038    CMake Error in CMakeLists.txt:
     1039      HIP_ARCHITECTURES is empty for target
     1040      "directed_tests.runtimeApi.cooperativeGrps.grid_group_data_sharing".
     1041    
     1042    
  >> 1043    CMake Error in CMakeLists.txt:
     1044      HIP_ARCHITECTURES is empty for target
     1045      "directed_tests.runtimeApi.cooperativeGrps.multi_grid_group_all_gpus".
     1046    
     1047    
  >> 1048    CMake Error in CMakeLists.txt:
     1049      HIP_ARCHITECTURES is empty for target
     1050      "directed_tests.runtimeApi.cooperativeGrps.hipLaunchCoopMultiKernel".
     1051    
     1052    
  >> 1053    CMake Error in CMakeLists.txt:
     1054      HIP_ARCHITECTURES is empty for target
     1055      "directed_tests.runtimeApi.cooperativeGrps.hipCGGridGroupTypeViaPublicApi".
     1056    
     1057    
  >> 1058    CMake Error in CMakeLists.txt:
     1059      HIP_ARCHITECTURES is empty for target
     1060      "directed_tests.runtimeApi.cooperativeGrps.hipLaunchCooperativeKernel".
     1061    
     1062    
  >> 1063    CMake Error in CMakeLists.txt:
     1064      HIP_ARCHITECTURES is empty for target
     1065      "directed_tests.runtimeApi.cooperativeGrps.simple_multi_grid_group_barrier".
     1066    
     1067    
  >> 1068    CMake Error in CMakeLists.txt:
     1069      HIP_ARCHITECTURES is empty for target
     1070      "directed_tests.runtimeApi.cooperativeGrps.api_failure_tests".
     1071    
     1072    
  >> 1073    CMake Error in CMakeLists.txt:
     1074      HIP_ARCHITECTURES is empty for target
     1075      "directed_tests.runtimeApi.cooperativeGrps.multi_gpu_streams".
     1076    
     1077    
  >> 1078    CMake Error in CMakeLists.txt:
     1079      HIP_ARCHITECTURES is empty for target
     1080      "directed_tests.runtimeApi.cooperativeGrps.hipCGGridGroupType".
     1081    
     1082    
  >> 1083    CMake Error in CMakeLists.txt:
     1084      HIP_ARCHITECTURES is empty for target
     1085      "directed_tests.runtimeApi.cooperativeGrps.hipCGGridGroupTypeViaBaseType".
     1086    
     1087    
  >> 1088    CMake Error in CMakeLists.txt:
     1089      HIP_ARCHITECTURES is empty for target
     1090      "directed_tests.runtimeApi.cooperativeGrps.hipCGThreadBlockType".
     1091    
     1092    
  >> 1093    CMake Error in CMakeLists.txt:
     1094      HIP_ARCHITECTURES is empty for target
     1095      "directed_tests.runtimeApi.cooperativeGrps.hipCGThreadBlockTypeViaPublicApi".
     1096    
     1097    
  >> 1098    CMake Error in CMakeLists.txt:
     1099      HIP_ARCHITECTURES is empty for target
     1100      "directed_tests.runtimeApi.cooperativeGrps.hipCGThreadBlockTypeViaBaseType".
     1101    
     1102    
  >> 1103    CMake Error in CMakeLists.txt:
     1104      HIP_ARCHITECTURES is empty for target
     1105      "directed_tests.runtimeApi.cooperativeGrps.hipCGMultiGridGroupTypeViaBaseType".
     1106    
     1107    
  >> 1108    CMake Error in CMakeLists.txt:
     1109      HIP_ARCHITECTURES is empty for target
     1110      "directed_tests.runtimeApi.cooperativeGrps.hipCGMultiGridGroupTypeViaPublicApi".
     1111    
     1112    
  >> 1113    CMake Error in CMakeLists.txt:
     1114      HIP_ARCHITECTURES is empty for target
     1115      "directed_tests.runtimeApi.device.hipDeviceGetByPCIBusId".
     1116    
     1117    
  >> 1118    CMake Error in CMakeLists.txt:
     1119      HIP_ARCHITECTURES is empty for target
     1120      "directed_tests.runtimeApi.device.hipSetCachceConfig".
     1121    
     1122    
  >> 1123    CMake Error in CMakeLists.txt:
     1124      HIP_ARCHITECTURES is empty for target
     1125      "directed_tests.runtimeApi.device.hipDeviceComputeCapability".
     1126    
     1127    
  >> 1128    CMake Error in CMakeLists.txt:
     1129      HIP_ARCHITECTURES is empty for target
     1130      "directed_tests.runtimeApi.device.hipDeviceGetPCIBusId".
     1131    
     1132    
  >> 1133    CMake Error in CMakeLists.txt:
     1134      HIP_ARCHITECTURES is empty for target
     1135      "directed_tests.runtimeApi.device.hipDeviceGetName".
     1136    
     1137    
  >> 1138    CMake Error in CMakeLists.txt:
     1139      HIP_ARCHITECTURES is empty for target
     1140      "directed_tests.runtimeApi.device.hipChooseDevice".
     1141    
     1142    
  >> 1143    CMake Error in CMakeLists.txt:
     1144      HIP_ARCHITECTURES is empty for target
     1145      "directed_tests.runtimeApi.device.hipDeviceSynchronize".
     1146    
     1147    
  >> 1148    CMake Error in CMakeLists.txt:
     1149      HIP_ARCHITECTURES is empty for target
     1150      "directed_tests.runtimeApi.device.hipDeviceTotalMem".
     1151    
     1152    
  >> 1153    CMake Error in CMakeLists.txt:
     1154      HIP_ARCHITECTURES is empty for target
     1155      "directed_tests.runtimeApi.device.hipGetDeviceAttribute".
     1156    
     1157    
  >> 1158    CMake Error in CMakeLists.txt:
     1159      HIP_ARCHITECTURES is empty for target
     1160      "directed_tests.runtimeApi.device.hipGetDeviceProperties".
     1161    
     1162    
  >> 1163    CMake Error in CMakeLists.txt:
     1164      HIP_ARCHITECTURES is empty for target
     1165      "directed_tests.runtimeApi.device.hipRuntimeGetVersion".
     1166    
     1167    
  >> 1168    CMake Error in CMakeLists.txt:
     1169      HIP_ARCHITECTURES is empty for target
     1170      "directed_tests.runtimeApi.device.hipSetDeviceFlags".
     1171    
     1172    
  >> 1173    CMake Error in CMakeLists.txt:
     1174      HIP_ARCHITECTURES is empty for target
     1175      "directed_tests.runtimeApi.device.hipSetGetDevice".
     1176    
     1177    
  >> 1178    CMake Error in CMakeLists.txt:
     1179      HIP_ARCHITECTURES is empty for target
     1180      "directed_tests.runtimeApi.error.hipPeekAtLastError".
     1181    
     1182    
  >> 1183    CMake Error in CMakeLists.txt:
     1184      HIP_ARCHITECTURES is empty for target
     1185      "directed_tests.runtimeApi.event.hipEventIpc".
     1186    
     1187    
  >> 1188    CMake Error in CMakeLists.txt:
     1189      HIP_ARCHITECTURES is empty for target
     1190      "directed_tests.runtimeApi.event.record_event".
     1191    
     1192    
  >> 1193    CMake Error in CMakeLists.txt:
     1194      HIP_ARCHITECTURES is empty for target
     1195      "directed_tests.runtimeApi.event.hipEventMultiThreaded".
     1196    
     1197    
  >> 1198    CMake Error in CMakeLists.txt:
     1199      HIP_ARCHITECTURES is empty for target
     1200      "directed_tests.runtimeApi.event.hipEvent".
     1201    
     1202    
  >> 1203    CMake Error in CMakeLists.txt:
     1204      HIP_ARCHITECTURES is empty for target
     1205      "directed_tests.runtimeApi.event.hipEventElapsedTime".
     1206    
     1207    
  >> 1208    CMake Error in CMakeLists.txt:
     1209      HIP_ARCHITECTURES is empty for target
     1210      "directed_tests.runtimeApi.event.hipEventRecord".
     1211    
     1212    
  >> 1213    CMake Error in CMakeLists.txt:
     1214      HIP_ARCHITECTURES is empty for target
     1215      "directed_tests.runtimeApi.occupancy.hipOccupancyMaxActiveBlocksPerMultiprocessor".
     1216    
     1217    
  >> 1218    CMake Error in CMakeLists.txt:
     1219      HIP_ARCHITECTURES is empty for target
     1220      "directed_tests.runtimeApi.occupancy.hipOccupancyMaxPotentialBlockSize".
     1221    
     1222    
  >> 1223    CMake Error in CMakeLists.txt:
     1224      HIP_ARCHITECTURES is empty for target
     1225      "performance_tests.dispatch.hipPerfDispatchSpeed".
     1226    
     1227    
  >> 1228    CMake Error in CMakeLists.txt:
     1229      HIP_ARCHITECTURES is empty for target
     1230      "performance_tests.compute.hipPerfDotProduct".
     1231    
     1232    
  >> 1233    CMake Error in CMakeLists.txt:
     1234      HIP_ARCHITECTURES is empty for target
     1235      "performance_tests.compute.hipPerfMandelbrot".
     1236    
     1237    
  >> 1238    CMake Error in CMakeLists.txt:
     1239      HIP_ARCHITECTURES is empty for target
     1240      "performance_tests.memory.hipPerfSharedMemReadSpeed".
     1241    
     1242    
  >> 1243    CMake Error in CMakeLists.txt:
     1244      HIP_ARCHITECTURES is empty for target
     1245      "performance_tests.memory.hipPerfSampleRate".
     1246    
     1247    
  >> 1248    CMake Error in CMakeLists.txt:
     1249      HIP_ARCHITECTURES is empty for target
     1250      "performance_tests.memory.hipPerfMemset".
     1251    
     1252    
  >> 1253    CMake Error in CMakeLists.txt:
     1254      HIP_ARCHITECTURES is empty for target
     1255      "performance_tests.memory.hipPerfMemcpy".
     1256    
     1257    
  >> 1258    CMake Error in CMakeLists.txt:
     1259      HIP_ARCHITECTURES is empty for target
     1260      "performance_tests.memory.hipPerfMemMallocCpyFree".
     1261    
     1262    
  >> 1263    CMake Error in CMakeLists.txt:
     1264      HIP_ARCHITECTURES is empty for target
     1265      "performance_tests.memory.hipPerfDevMemWriteSpeed".
     1266    
     1267    
  >> 1268    CMake Error in CMakeLists.txt:
     1269      HIP_ARCHITECTURES is empty for target
     1270      "performance_tests.memory.hipPerfBufferCopyRectSpeed".
     1271    
     1272    
  >> 1273    CMake Error in CMakeLists.txt:
     1274      HIP_ARCHITECTURES is empty for target
     1275      "performance_tests.memory.hipPerfBufferCopySpeed".
     1276    
     1277    
  >> 1278    CMake Error in CMakeLists.txt:
     1279      HIP_ARCHITECTURES is empty for target
     1280      "performance_tests.memory.hipPerfDevMemReadSpeed".
     1281    
     1282    
  >> 1283    CMake Error in CMakeLists.txt:
     1284      HIP_ARCHITECTURES is empty for target
     1285      "performance_tests.stream.hipPerfStreamCreateCopyDestroy".
     1286    
     1287    
  >> 1288    CMake Error in CMakeLists.txt:
     1289      HIP_ARCHITECTURES is empty for target
     1290      "performance_tests.stream.hipPerfStreamConcurrency".
     1291    
     1292    
  >> 1293    CMake Error in CMakeLists.txt:
     1294      HIP_ARCHITECTURES is empty for target
     1295      "performance_tests.stream.hipPerfDeviceConcurrency".
     1296    
     1297    
     1298    -- Generating done
     1299    CMake Generate step failed.  Build files cannot be regenerated correctly.

See build log for details:
  /tmp/user/spack-stage/spack-stage-hip-4.3.1-tlzwetxef5vw4roytm6wdhoeep3m5vcw/spack-build-out.txt
```

--- a/var/spack/repos/builtin/packages/hip/package.py
+++ b/var/spack/repos/builtin/packages/hip/package.py
@@ -71,7 +71,7 @@ class Hip(CMakePackage):
     patch('0004-Drop-clang-rt-builtins-linking-on-hip-host.3.10.0.patch', when='@3.10.0:4.1.0')
 
     # Tests are broken when using cmake 3.21
-    with when('^cmake@3.21.0:3.21.2'):
+    with when('^cmake@3.21.0:'):
         patch('0005-Disable-tests-3.5.0.patch', when='@3.5.0')
         patch('0005-Disable-tests-3.6.0.patch', when='@3.6.0:3.8.0')
         patch('0005-Disable-tests-3.9.0.patch', when='@3.9.0:4.0.0')

